src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -278,116 +304,3125 @@ static ColliderJntSphInit D_80907014 = {
         OC2_TYPE_1,
         COLSHAPE_JNTSPH,
     },
-    2,
-    D_80906FCC,
+    ARRAY_COUNT(sJntSphItemsInit2),
+    sJntSphItemsInit2,
+};
+
+static Color_RGBA8 sPrimColor = { 0, 120, 0, 255 };
+
+static Color_RGBA8 sEnvColor = { 0, 120, 0, 255 };
+
+static Vec3f D_8090702C[] = {
+    { 10.0f, -10.0f, 0.0f },
+    { 0.0f, 0.0f, -60.0f },
+    { 70.0f, -30.0f, 10.0f },
+};
+
+static Vec3f D_80907050[] = {
+    { -20.0f, 0.0f, 0.0f },
+    { -15.0f, 0.0f, 10.0f },
+    { -16.0f, -12.0f, 40.0f },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_808FCF40.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_808FCF5C.s")
+static s16 D_80907074[] = { 0, 1, 2, 2, 1, 0 };
+
+static s16 D_80907080 = 0;
+
+static u8 D_80907084[] = { 0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20 };
+
+static u8 D_80907090[] = { 1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21 };
+
+static s8 D_8090709C[] = {
+    0xFF, 0xFF, 0x01, 0xFF, 0x03, 0x04, 0xFF, 0xFF, 0x05, 0xFF, 0x06, 0x07, 0xFF, 0xFF, 0x08, 0xFF,
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+    0x00, 0xFF, 0xFF, 0x02, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0xFF, 0xFF, 0xFF, 0x0F, 0xFF, 0x00,
+};
+
+static s8 D_809070CC[] = {
+    0xFF, 0xFF, 0x01, 0xFF, 0x03, 0x04, 0xFF, 0xFF, 0x05, 0xFF, 0x06, 0x07, 0xFF, 0xFF, 0x08, 0xFF,
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+    0x00, 0xFF, 0xFF, 0x02, 0x0C, 0x0D, 0x0E, 0x09, 0x0A, 0x0B, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x00,
+};
+
+static Vec3f D_809070FC = { 0.0f, 0.0f, 5000.0f };
+
+static Vec3f D_80907108 = { 0.0f, 2000.0f, 0.0f };
+
+static Vec3f D_80907114 = { 0.0f, 2000.0f, 0.0f };
+
+static Vec3f D_80907120 = { 0.0f, 0.0f, 17000.0f };
+
+static Vec3f D_8090712C = { 0.0f, 0.0f, 3000.0f };
+
+static Vec3f D_80907138 = { 0.0f, 0.0f, 0.0f };
+
+static s8 D_80907144[] = {
+    0xFF, 0xFF, 0x01, 0xFF, 0x03, 0x04, 0x05, 0xFF, 0x06, 0x07, 0x08, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+    0xFF, 0xFF, 0xFF, 0x02, 0x0C, 0x0D, 0x0E, 0x09, 0x0A, 0x0B, 0xFF, 0xFF, 0xFF, 0xFF, 0x00, 0x00,
+};
+
+static Vec3f D_80907164 = { 800.0f, 420.0f, 100.0f };
+
+static void* sEyeTextures[] = {
```

Comment:
We dont require segment addresses in arrays to be defined like this (though MM does) but since you already have them like this you can just leave them.

Reply:
What part of it? The `void*[]`? I typed it like that because it was an array of textures. Should I have done something else?

Reply:
usually what is done is we just leave the raw pointer.
`void* tex[] = {0x06001234, 0x06001233,etc...};`

Reply:
Ah gotcha. I'll probably leave them `extern`ed since I know what type they are and left a comment next to them and you said it's fine, but if wanted I can change it.